app/src/main/java/no/usn/mob3000/data/repository/auth/AuthRepository.kt

```diff
@@ -311,6 +313,7 @@ class AuthRepository(
             ) as T
 
             dto is ProfileDto && T::class == DomainUserProfile::class -> DomainUserProfile(
+                userId = dto.userId,
```

Comment:
I question this addition when there's already handling for the ID, but I'll let it slide.